internal/databases/mongo/oplog/applier.go

```diff
@@ -0,0 +1,67 @@
+package oplog
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/wal-g/wal-g/internal"
+
+	"github.com/wal-g/tracelog"
+)
+
+type Applier interface {
+	Apply(context.Context, chan Record, *sync.WaitGroup) (chan struct{}, chan error, error)
+}
+
+type DBApplier struct {
+	uri string
+	mc  *internal.MongoClient
+}
+
+func NewDBApplier(uri string) *DBApplier {
+	return &DBApplier{uri: uri}
+}
+
+func (dba *DBApplier) Apply(ctx context.Context, ch chan Record, wg *sync.WaitGroup) (chan struct{}, chan error, error) {
+	mc, err := internal.NewMongoClient(ctx, dba.uri)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if _, err := mc.GetOplogCollection(ctx); err != nil {
+		return nil, nil, err
+	}
+
+	done := make(chan struct{})
+	errc := make(chan error)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		defer mc.Close(ctx)
+		defer close(errc)
+		defer close(done)
+		for {
+			select {
+			case op, ok := <-ch:
+				tracelog.DebugLogger.Printf("Applyer receieved op %s (%s on %s), open: %t", op.TS, op.OP, op.NS, ok)
+
+				if !ok {
+					done <- struct{}{}
```

Comment:
This will stuck until someone read from chan. Is it intended behavior?
Maybe you want to use buffered chan?

Reply:
Yes it's "work is done, no error acquired" indicator.

Reply:
But you are doing it in a loop for bunch of works?

Reply:
Oh, I see, it's not a loop, actually